features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/Listing/Edit.jsx

```diff
@@ -0,0 +1,305 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useReducer, useState, useEffect } from 'react';
+import { FormattedMessage, injectIntl, useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import Icon from '@material-ui/core/Icon';
+import Alert from 'AppComponents/Shared/Alert';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import Dialog from '@material-ui/core/Dialog';
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+import { makeStyles } from '@material-ui/core/styles';
+import ServiceCatalog from 'AppData/ServiceCatalog';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemText from '@material-ui/core/ListItemText';
+
+const useStyles = makeStyles((theme) => ({
+    mandatoryStar: {
+        color: theme.palette.error.main,
+        marginLeft: theme.spacing(0.1),
+    },
+}));
+
+/**
+ * Reducer
+ * @param {JSON} state State.
+ * @returns {Promise} Promised state.
+ */
+function reducer(state, { field, value }) {
+    switch (field) {
+        case 'name':
+            return { ...state, [field]: value, displayName: value };
+        case 'serviceUrl':
+            return { ...state, [field]: value };
+        case 'definitionType':
+            return { ...state, [field]: value };
+        default:
+            return state;
+    }
+}
+
+/**
+* Service Catalog service edit
+* @param {any} props Props for edit function.
+* @returns {any} Returns the rendered UI for service edit.
+*/
+function Edit(props) {
+    const intl = useIntl();
+    const classes = useStyles();
+    const { getData, dataRow } = props;
+    const [open, setOpen] = useState(false);
+    const [schemaTypeList, setSchemaTypeList] = useState([]);
+    const [serviceTypeList, setServiceTypeList] = useState([]);
+    const toggleOpen = () => {
+        setOpen(!open);
+    };
+    const [initialState, setInitialState] = useState(dataRow);
+
+    const [state, dispatch] = useReducer(reducer, initialState);
+
+    const {
+        id,
+        name,
+        serviceUrl,
+        definitionType,
+    } = state;
+
+    useEffect(() => {
+        setInitialState(dataRow);
+        const settingPromise = ServiceCatalog.getSettings();
+        // Take from settings when backend API is implemented
+        settingPromise.then(() => {
+            setSchemaTypeList(['OAS2', 'OAS3', 'WSDL1', 'WSDL2', 'GRAPHQL_SDL', 'ASYNC_API']);
```

Comment:
Can't we extract mock values from OAS definition, Or shall we move this to API data level so that we know once the REST API is ready it's only that API file we need to change in order to wire the actual API

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9407/commits/532eb141892926576a6141bd54892204ae4af1b1